Sources/Leaf/LeafEncoder.swift

```diff
@@ -136,8 +136,9 @@ extension LeafEncoder {
     }
 
     private final class KeyedContainerImpl<Key>: KeyedEncodingContainerProtocol, LeafEncodingResolvable where Key: CodingKey {
-        private let encoder: EncoderImpl
+        private weak var encoder: EncoderImpl!
```

Comment:
Let's make this optional since we unwrap it anyway. That provides a fatal error with a message rather than just crashing (and won't complain when we roll out swiftlint)